echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/api/Notification.groovy

```diff
@@ -50,4 +54,40 @@ class Notification {
     NORMAL,
     HIGH
   }
+
+  /**
+   * Allows Spinnaker services sending Notifications through Echo to specify one or more interactive actions
+   * that, when acted upon by a user, cause a callback to Echo which gets routed to that originating service.
+   */
+  static class InteractiveActions {
+    String callbackServiceId
+    String callbackMessageId
+    String color = '#cccccc'
```

Comment:
Does `color` need to be a property on `InteractiveActions`?

Reply:
I was loosely basing the design on the Slack API, and they associate the color with each "attachment" (a.k.a. "block") of a message -- it basically controls the color of the vertical bar to the left of each block/part of the message. Since we only support a single attachment (it was already like this before my change), I figured it would go with `InteractiveActions` which eventually gets translated into the attachment.

Reply:
I think for something like this, having a `Map<String, String> attributes` might be better?

Reply:
I didn’t want to do that because I need to enforce this generic contract for any notification service implementation. That’s one of the requirements: they need to be able to provide this `serviceId` and `messageId` while handling the notification callback. Or did you mean about the `color` field only?

Reply:
Just the `color` field.

Reply:
OK. Let's refactor that some other time! 👍